tutorials/mnist_saliency_tutorial.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+import keras
+from keras.models import Sequential
+from keras.layers import Dense, Dropout, Activation, Flatten
+from keras.layers import Convolution2D, MaxPooling2D
+
+import tensorflow as tf
+from tensorflow.python.platform import flags
+
+from cleverhans.utils_mnist import data_mnist, model_mnist
+from cleverhans.utils_keras import accuracy
+from cleverhans.attacks import saliency
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('nb_epochs', 1, 'Number of epochs to train model')
+flags.DEFINE_integer('batch_size', 128, 'Size of training batches')
+flags.DEFINE_integer('nb_classes', 10, 'Number of classification classes')
+flags.DEFINE_integer('img_rows', 28, 'Input row dimension')
+flags.DEFINE_integer('img_cols', 28, 'Input column dimension')
+flags.DEFINE_integer('nb_filters', 64, 'Number of convolutional filter to use')
+flags.DEFINE_integer('nb_pool', 2, 'Size of pooling area for max pooling')
+
+
+def main(argv=None):
+    # Image dimensions ordering should follow the Theano convention
+    if keras.backend.image_dim_ordering() != 'th':
+        keras.backend.set_image_dim_ordering('th')
+        print "INFO: '~/.keras/keras.json' sets 'image_dim_ordering' to 'tf', temporarily setting to 'th'"
+
+    # Create TF session and set as Keras backend session
+    sess = tf.Session()
+    keras.backend.set_session(sess)
+    print "Created TensorFlow session and set Keras backend."
+
+    # Get MNIST test data
+    X_train, Y_train, X_test, Y_test = data_mnist()
+    print "Loaded MNIST test data."
+
+    # build the Keras model, and the
+    x = tf.placeholder(tf.float32, shape=(None, 1, 28, 28))
+    model, f_x = model_mnist(logits=True, input_ph=x)
+    model.compile(loss='categorical_crossentropy',
```

Comment:
Do we know if this uses `from_logits = True` when computing the cross-entropy? The default when you call the `categorical_crossentropy` directly is `False` and that caused trouble for `fgsm` because it is not very numerically accurate.